presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -483,13 +493,20 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
             // Create new projections for the new join clauses
             List<JoinNode.EquiJoinClause> equiJoinClauses = new ArrayList<>();
             ImmutableList.Builder<Expression> joinFilterBuilder = ImmutableList.builder();
+            ImmutableList.Builder<ComparisonExpression> dynamicFilteringClauses = ImmutableList.builder();
             for (Expression conjunct : extractConjuncts(newJoinPredicate)) {
-                if (joinEqualityExpression(conjunct, node.getLeft().getOutputSymbols(), node.getRight().getOutputSymbols())) {
```

Comment:
let's encapsulate non-equi DF code in `createDynamicFilters` method. Then these code changes are not needed

Reply:
Sounds good - I've changed `createDynamicFilters` signature to accept both `equiJoinClauses` and `joinFilterBuilder.build()`.